src/Avalonia.Controls/DateTimePickers/LoopingPanel.cs

```diff
@@ -0,0 +1,259 @@
+
+            var itemWid = finalSize.Width;
+            var itemHgt = _owner.ItemHeight;
+            var initY = (finalSize.Height / 2.0) - (itemHgt / 2.0);
+            var children = Children;
+            var offY = Offset.Y;
+
+            //When not looping, currentSet will always be 0
+            //When looping, we measure for 10x _owner.ItemCount, so we need to figure out
+            //which "set" of items we're in based on the offset so we know where to properly
+            //place items
+            var singleExtent = _owner.ItemCount * itemHgt;
+            var currentSet = Math.Truncate(offY / singleExtent);
+
+            int selIndex = _owner.SelectedIndex;
+            selIndex = selIndex == -1 ? 0 : selIndex;
+
+            //When looping, the selected item should always be the middle item, equivalent in index
+            //to _totalItemsInViewport
+            //When not looping, if we're near the beginning of the list, our first item may be less than
+            //_totalItemsInViewport, so we need to make sure in that case to make the selected item, the
+            //actual selected index
+            var childIndexOfSelected = _totalItemsInViewport;
+            if (!_owner.ShouldLoop && selIndex < _totalItemsInViewport)
+                childIndexOfSelected = selIndex;
+
+            //Our selected container forms our "anchor" and all other containers are placed around this
+            IControl containerOfSelected = children[childIndexOfSelected];
+
+            //Then we need to know how many containers are above and below the selected item
+            //Should be _totalItemsInViewport for both, unless not looping & near items start
+            //# containers above is just the index of the selected item container
+            var numContainersAboveSelected = childIndexOfSelected;
+
+            //Move initY to where we actually want to start placing the items
+            initY += (singleExtent * currentSet) + (selIndex * itemHgt);
+
+            //We first arrange the selected item
+            Rect rc = new Rect(0, initY - offY, itemWid, itemHgt);
+            containerOfSelected.Arrange(rc);
+
+            //Arrange all items above
+            var prevY = initY - itemHgt;
+            for (int i = numContainersAboveSelected - 1; i >= 0; i--)
+            {
+                rc = new Rect(0, prevY - offY, itemWid, itemHgt);
+                children[i].Arrange(rc);
+                prevY -= itemHgt;
+            }
+
+            //Finally arrange all items below
+            var nextY = initY + itemHgt;
+            for (int i = childIndexOfSelected + 1; i < children.Count; i++)
+            {
+                rc = new Rect(0, nextY - offY, itemWid, itemHgt);
+                children[i].Arrange(rc);
+                nextY += itemHgt;
+            }
+
+            return new Size(itemWid, _extent.Height);
+        }
+
+        public bool CanHorizontallyScroll { get => false; set => _ = value; }
```

Comment:
this is a bit weird, why can't this be a getter only?

Reply:
CanHorizontallyScroll and CanVerticallyScroll are read/write in ILogicalScrollable, and it wouldn't let me just use the getter. I could be mistaken, but I believe the ScrollViewer still tries to set these even with logical scroll enabled, so I just discard it. I agree, it is a bit weird tho.